src/html.cr

```diff
@@ -34,42 +36,27 @@ module HTML
     end
   end
 
-  # Returns a string where some named and all numeric character references
+  # Returns a string where named and numeric character references
   # (e.g. &gt;, &#62;, &x3e;) in *string* are replaced with the corresponding
-  # unicode characters. Only these named entities are replaced:
-  # apos, amp, quot, gt, lt and nbsp.
+  # unicode characters.
   #
   # ```
   # HTML.unescape("Crystal &amp; You") # => "Crystal & You"
   # ```
   def self.unescape(string : String) : String
-    return string unless string.includes? '&'
-
-    string.gsub(/&(apos|amp|quot|gt|lt|nbsp|\#[0-9]+|\#[xX][0-9A-Fa-f]+);/) do |string, _match|
-      match = _match[1]
-      case match
-      when "apos" then "'"
-      when "amp"  then "&"
-      when "quot" then "\""
-      when "gt"   then ">"
-      when "lt"   then "<"
-      when "nbsp" then " "
-      when /\A#0*(\d+)\z/
-        n = $1.to_i
-        if n <= Char::MAX_CODEPOINT
-          n.unsafe_chr
-        else
-          "&##{$1};"
-        end
-      when /\A#x([0-9a-f]+)\z/i
-        n = $1.to_i(16)
-        if n <= Char::MAX_CODEPOINT
-          n.unsafe_chr
-        else
-          "&#x#{$1};"
-        end
+    string.gsub(/&(?:([a-zA-Z]+;?)|\#([0-9]+);?|\#[xX]([0-9A-Fa-f]+);?)/) do |string, match|
+      if code = match[1]?
+        HTML::SINGLE_CHAR_ENTITIES[code]? ||
+          HTML::DOUBLE_CHAR_ENTITIES[code]? ||
```

Comment:
Do we really need to seperate these hashes? They're `:nodoc:` anyway so why not make it 1 hash and 1 hash lookup?

Reply:
I think it occupies more size in memory. Chars are just four bytes, while a string is minimum 9 bytes plus an indirection. Go implements it like that too and I think it's fine. In any case, the slowness comes from the regex.

Reply:
It might be faster and more space efficient to encode the double char entities with an unused unicode section and use the section index minus offset to retrieve the actual strings from a tuple or array instead of two hash lookups.

Reply:
Please send a PR with optimizations later. This is just about getting it right at a reasonable performance.